libstuff/SQResult.cpp

```diff
@@ -76,3 +76,32 @@ bool SQResult::deserialize(const string& json) {
     clear();
     return false;
 }
+
+bool SQResult::empty() const {
+    return rows.empty();
+}
+
+size_t SQResult::size() const {
+    return rows.size();
+}
+
+void SQResult::clear() {
+    headers.clear();
+    rows.clear();
+}
+
+vector<string>& SQResult::operator[](size_t rowNum) {
+    try {
+        return rows.at(rowNum);
+    } catch (const out_of_range& e) {
+        STHROW("Out of range");
```

Comment:
why not just let the standard exception float up?

Reply:
Because if this happens in an auth command this gets handled in a way that makes sense and causes the command to fail. We don't have explicit handling up the stack for `std::out_of_range`.

Reply:
yeah that makes sense